### PR TITLE
Attribute variance

### LINKED ISSUE #153: Implement attribute "var" (variance)

### LINKED ISSUE #156: Implement attribute "sumpos" (sum of positive values)

### LINKED ISSUE #157: Implement attribute "sumneg" (sum of negative values)

### LINKED ISSUE #158: Implement attribute "meanabs" (mean of absolute values)

### LINKED ISSUE #159: Implement attribute "meanpos" (mean of positive values)

### LINKED ISSUE #160: Implement attribute "meanneg" (mean of negative values)

### PR DESCRIPTION
**Pull request by AdilKhurshid**
closes #160
closes #159
closes #158 
closes #157 
closes #156 
closes #153

### CODE REVIEW

File: internal/vds/vds.go [hunk 1]
```diff
@@ -171,9 +171,20 @@ func GetAttributeType(attribute string) (int, error) {
 	case "rms":         return C.RMS,    nil
 	case "var":         return C.VAR,    nil
 	case "sd":          return C.SD,     nil
+	case "sumpos":      return C.SUMPOS, nil
 	case "":            fallthrough
 	default:
-		options := "samplevalue, min, max, maxabs, mean, median, rms, var, sd"
+		options := `
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/vds/vds.go]: I tried to fix the issue. I don't know whether it's the best way of solving this kind of issue or not. Just suggest me if you know a better way to fix this issue. :) 
https://github.com/equinor/vds-slice/pull/164/files#diff-755991db743271ea57cbf696bc130f892177693a96014841f30c707532f91765R181-R183

File: docs/attribute.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ rms         | Root mean square
 var         | Variance
 sd          | Standard deviation
 sumpos      | Sum of positive samples
+sumneg      | Sum of Negative samples
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply docs/attribute.md]: fixed.
https://github.com/equinor/vds-slice/pull/164/files#diff-97a3f4c45e3c39062800139ceef4b71f6ebea186020e3cb1768e15749f36e312R28-R37

File: internal/vds/vds_test.go [hunk 3]
```diff
@@ -900,16 +900,39 @@ func TestSurfaceHorizontalBounds(t *testing.T) {
...
+		{  1,         1.5,        0,         6.5,      fill,  0,         fill, fill }, // meanpos
+		{ -1.5,      -1,         -8.5,       0,        fill, -16.5,      fill, fill }, // meanneg
 		{ -0.5,       0.5,       -8.5,       6.5,      fill, -16.5,      fill, fill }, // median
 		{  1.5,       1.5,        8.958237,  7.0887237,fill,  17.442764, fill, fill }, // rms
+		{  2,           2,        8,         8,        fill,  32,        fill, fill }, // var
 		{  1.4142135, 1.4142135,  2.828427,  2.828427, fill,   5.656854, fill, fill }, // sd
+		{  2,         4.5,        0,         32.5,     fill,   0,        fill, fill }, // sumpos
+		{ -4.5,      -2,        -42.5,       0,        fill, -82.5,      fill, fill }, // sumneg
+
+
```
[the remark this answers is not in the document]
Comment (achaikou) [reply internal/vds/vds_test.go:935]: You reacted with :+1: , but did not fix it :)

[If you don't want to do it is fine, but then maybe just reply with something like "nah" so I am not confused :) 
I consider :+1: to be "yep, agree, will fix". So if I see situation like this, I assume that author wanted to fix, but simply forgot, so I am reminding again :) ]

[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/vds/vds_test.go:935]: Sorry, 🫣 my mistake. I totally forgot to add this change. I will fix this. Thanks

File: internal/vds/vds.go [hunk 4]
```diff
@@ -162,17 +162,25 @@ func GetInterpolationMethod(interpolation string) (int, error) {
...
+	case "var":         return C.VAR,     nil
+	case "sd":          return C.SD,      nil
+	case "sumpos":      return C.SUMPOS,  nil
+	case "sumneg":      return C.SUMNEG,  nil
 	case "":            fallthrough
 	default:
-		options := "samplevalue, min, max, maxabs, mean, median, rms, sd"
+		options := "samplevalue," + " min,"    + " max,"    + " maxabs," + 
+		" mean," + " meanabs"     + " meanpos" + " meanneg" + " median," +  
+		" rms,"  + " var,"        + " sd,"     + " sumpos"  + " sumneg"
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/vds/vds.go:183]: Thanks, nice suggestion. I will add this change. 😇

**Review by achaikou — COMMENTED** [review]
Looks good!
Rebase and apply fixups and we can merge!

(Though look at the "    Add mean of negative values attribute" commit. My tool shows weird spaces before commit name.)

**Comment by AdilKhurshid** [comment]
> Looks good! Rebase and apply fixups and we can merge!
> 
> (Though look at the " Add mean of negative values attribute" commit. My tool shows weird spaces before commit name.)

I have fixed this issue. Just check if it's fine and approves. 😊

**Pull request merged**